### PR TITLE
Cleanup includes fwd decls

### PR DESCRIPTION
**Pull request by endJunction**
- One large commit, unfortunately, forcing full include paths in FileIO, which also cleanups them and use forward declarations where possible.
 - Remove unused zLibDataCompressor.
 - Some include related changes here and there.

I tried to use same scheme for the includes:
 - class declarations (if in the cpp file)
 - system (stl and alike includes)
 - our third party lib includes
 - alphabetically BaseLib, GeoLib etc.
 - other local includes.

~~The compilation times did not change unfortunately.~~
__Update:__
With the second batch of changes there compile time is reduced by ~4% relative to the ufz/master.

### CODE REVIEW

File: FileIO/FEFLOWInterface.cpp [hunk 1]
```diff
@@ -9,24 +9,25 @@
...
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/FileTools.h"
+#include "BaseLib/StringTools.h"
+
+#include "GeoLib/GEOObjects.h"
 #include "GeoLib/Point.h"
+#include "GeoLib/Polygon.h"
+
+#include "MeshLib/Elements/Hex.h"
```
Comment (wenqing) [inline FileIO/FEFLOWInterface.cpp]: It might be better to put these mesh related includes, which occur multiply in different files,  in one header file.

**Comment by TomFischer** [comment]
Looks good. :fast_forward:

File: FileIO/FEFLOWInterface.cpp (hunk 1 above)
Comment (endJunction) [inline FileIO/FEFLOWInterface.cpp]: Created an Elements.h include for this case.

**Comment by norihiro-w** [comment]
:+1:

**Pull request merged**